pkg/restore/apimanager_restore.go

```diff
@@ -0,0 +1,418 @@
+package restore
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/component"
+	"github.com/3scale/3scale-operator/pkg/backup"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type APIManagerRestore struct {
+	options *APIManagerRestoreOptions
+}
+
+func NewAPIManagerRestore(options *APIManagerRestoreOptions) *APIManagerRestore {
+	return &APIManagerRestore{
+		options: options,
+	}
+}
+
+const (
+	RestorePVCMountPath           = "/backup"
+	SystemFileStoragePVCMountPath = "/system-filestorage-pvc"
+)
+
+var secretsToRestore map[string]string = map[string]string{
+	"SystemSMTP": "system-smtp",
+	"SystemSeed": "system-seed",
+	// "SystemDatabase":      "system-database", // We decided this is done by the user. Reason for that is that he might want to restore it in another place
```

Comment:
These type of comments are useful for the code?

Reply:
I think so. It explains why this subset of secrets/configmaps have not been applied